core/src/main/java/com/linecorp/armeria/internal/common/util/SslContextUtil.java

```diff
@@ -124,6 +124,9 @@ public static SslContext createSslContext(
                 }
             }
 
+            // Set endpoint identification algorithm so that JDK's default X509TrustManager implementation
+            // performs host name checks. This options is effective only for clients.
+            builder.endpointIdentificationAlgorithm("HTTPS");
```

Comment:
Netty added a new API to specify the identification algorithm, which we used to set via `SslHandler`
https://github.com/netty/netty/wiki/Netty-4.2-Migration-Guide#new-best-practices:~:text=Ensure%20endpoint%20validation,in%20Netty%204.1.112.